engine/launcher/src/main/scala/org/enso/launcher/InternalOpts.scala

```diff
@@ -0,0 +1,83 @@
+package org.enso.launcher
+
+import java.io.IOException
+import java.nio.file.{Files, NoSuchFileException, Path}
+
+import org.enso.cli.Opts
+import org.enso.cli.Opts.implicits._
+import cats.implicits._
+import org.enso.launcher.internal.OS
+import sys.process._
+
+object InternalOpts {
+  private def REMOVE_OLD_EXECUTABLE = "internal-remove-old-executable"
+
+  def topLevelOptions: Opts[Unit] = {
+    val removeOldExecutableOpt = Opts
+      .optionalParameter[Path](
+        REMOVE_OLD_EXECUTABLE,
+        "PATH",
+        "Removes the old executable file after the installation. " +
+        "Used as a workaround for file locking on Windows."
+      )
+      .hidden
+
+    removeOldExecutableOpt map {
+      case Some(oldExecutablePath) =>
+        removeOldExecutable(oldExecutablePath)
+        sys.exit(0)
+      case None =>
+    }
+  }
+
+  def runWithNewLauncher(pathToNewLauncher: Path): Runner =
+    new Runner(pathToNewLauncher)
+
+  class Runner private[InternalOpts] (pathToNewLauncher: Path) {
+    def removeOldExecutableAndExit(oldExecutablePath: Path): Nothing = {
+      val command = Seq(
+        pathToNewLauncher.toAbsolutePath.toString,
+        "--" + REMOVE_OLD_EXECUTABLE,
+        oldExecutablePath.toAbsolutePath.toString
+      )
+      runDetachedAndExit(command)
+    }
+  }
+
+  private def removeOldExecutable(oldExecutablePath: Path): Unit = {
+    @scala.annotation.tailrec
+    def tryDeleting(retries: Int): Unit = {
+      try {
+        Files.delete(oldExecutablePath)
+      } catch {
+        case _: NoSuchFileException =>
+        case e: IOException =>
+          if (retries > 0) {
+            e.printStackTrace()
+            Thread.sleep(500)
+            tryDeleting(retries - 1)
+          } else {
+            println(
+              s"Cannot delete old executable at $oldExecutablePath after " +
+              s"multiple retries. Please try removing it manually."
+            )
+            e.printStackTrace()
+          }
+      }
+    }
+
+    tryDeleting(30)
```

Comment:
That's 15 seconds. That's a lot. I don't want to wait 15 seconds, retry a smaller number of times and let me know you failed – it's a non-fatal failure

Reply:
But do you think that this should print a warning (which will most likely be missed in between the many test logs) or fail that test suite?

Reply:
Test suite should probably get failed. User should not get failed.

Reply:
I will add a message indicating the failure, but I'd like to keep that timeout - it is a short time considering that on Windows an antivirus may lock an application for an arbitrarily long amount of time. And it shouldn't affec the tests if everything works correctly.